takari-lifecycle-plugin/src/test/java/io/takari/maven/plugins/compile/jdt/CompileJdtClasspathTest.java

```diff
@@ -142,6 +145,29 @@ public void testRepository_classpathOrder() throws Exception {
     mojos.assertBuildOutputs(parent, "module-a/target/classes/reactor/modulea/ModuleA.class");
   }
 
+  @Test
+  public void testReactorUnused() throws Exception {
```

Comment:
Unused dependency check is implemented in `AbstractCompileMojo`, which is used by all compiler backends. For consistency, move the test to `CompileTest`. (alternatively, you can move the check implementation to CompilerJdt, in which case the test can stay here).